### PR TITLE
Fix Cluster shutdown handling of http connections

### PR DESCRIPTION
**Pull request by dothebart**
- make all parts of the simple http client depend on the 'CommunicationPhase'
- Assert whether we are in an open/close loop

### CODE REVIEW

**Comment by dothebart** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/2574/

**Comment by dothebart** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr-windows-onlycatch/549/ Fix scope of variable for windows build.

File: lib/SimpleHttpClient/SslClientConnection.cpp [hunk 1]
```diff
@@ -502,6 +504,9 @@ bool SslClientConnection::readClientConnection(StringBuffer& stringBuffer,
     switch (SSL_get_error(_ssl, lenRead)) {
       case SSL_ERROR_NONE:
         stringBuffer.increaseLength(lenRead);
+#ifdef ARANGODB_ENABLE_MAINTAINER_MODE
+	_read += lenRead;
```
Comment (jsteemann) [inline lib/SimpleHttpClient/SslClientConnection.cpp]: indentation is not right.

File: lib/SimpleHttpClient/GeneralClientConnection.cpp [hunk 2]
```diff
@@ -157,6 +167,16 @@ bool GeneralClientConnection::connect() {
 
 void GeneralClientConnection::disconnect() {
   if (isConnected()) {
+#ifdef ARANGODB_ENABLE_MAINTAINER_MODE
+    if ((_written + _read) == 0) {
+      std::string bt;
+      TRI_GetBacktrace(bt);
+      LOG_TOPIC(WARN, Logger::COMMUNICATION) <<
+        "Closing HTTP-connection right after opening it without sending data!" + bt;
```
Comment (jsteemann) [inline lib/SimpleHttpClient/GeneralClientConnection.cpp]:
```suggestion
        "Closing HTTP-connection right after opening it without sending data!" << bt;
```

**Pull request merged**